vms/evm/destination_client.go

```diff
@@ -144,17 +163,22 @@ func NewDestinationClient(
 				// Limit the number of transactions in the mempool for each account,
 				// otherwise they may be dropped.
 				txQueue := make(chan struct{}, poolTxsPerAccount)
+				messageChan := make(chan MessageData, 0)
+				messageChans[i] = messageChan
 				accountSigners[i] = accountSigner{
-					signer:       signer,
-					currentNonce: currentNonce,
-					lock:         &sync.Mutex{},
-					txQueue:      txQueue,
-				}
-				selectCases[i] = reflect.SelectCase{
-					Dir:  reflect.SelectSend,
-					Chan: reflect.ValueOf(txQueue),
-					Send: reflect.ValueOf(struct{}{}),
+					logger:            logger.With(zap.Stringer("address", signer.Address())),
```

Comment:
```suggestion
					logger:            logger.With(zap.Stringer("senderAddress", signer.Address())),
```
To disambiguate from the called address